service/src/main/java/com/codeforcommunity/processor/AuthProcessorImpl.java

```diff
@@ -90,4 +92,25 @@ private String getSignature(String token) {
         return token.split("\\.")[2];
     }
 
+    @Override
+    public VerifySecretKeyResponse validateSecretKey(String secretKey) {
+      int userId;
+      try {
+          userId = authDatabase.validateSecretKey(secretKey);
+      }
+      catch (AuthException e) {
+          return new VerifySecretKeyResponse().setUserId(-1).setMessage(e.getMessage());
```

Comment:
Why would we want this as the response? What's the point of returning a message and a userId? Why not just a boolean?

Reply:
Done